src/runtime/entitlements-manager.js

```diff
@@ -50,16 +51,61 @@ import {warn} from '../utils/log';
 
 const SERVICE_ID = 'subscribe.google.com';
 
+/**
+ * Properties:
+ *   - isClosable - determine whether the view is closable.
+ *   - onResult - callback to get the intervention result and decide if it completes.
+ *                Takes either a normal or async function and returns `true` if the
+ *                intervention should be marked complete.
+ *
+ * @typedef {{
+ *   isClosable: (boolean|undefined),
+ *   onResult: ((function(!Object):(Promise<Boolean>|Boolean))|undefined),
+ * }}
+ */
+export let ShowInterventionParams;
+
+class Intervention {
+  /** @public @const {string} */
+  type;
+  /** @public @const {string} */
+  configurationId;
+}
+
+export class AvailableIntervention extends Intervention {
+  /**
+   * @param {Intervention} original
+   * @param {!./deps.DepsDef} deps
+   */
+  constructor(original, deps) {
+    super();
+    Object.assign(this, original);
+    /** @private @const {!./deps.DepsDef} */
+    this.deps_ = deps;
+  }
+  /**
+   * Starts the intervention flow.
+   * @param {!ShowInterventionParams=} params
+   * @return {?}
```

Comment:
Does it return a Promise?

Reply:
In this case it would always be a promise. I left it generic because there might be situations in the future where a publisher provided intervention would not be async. Maybe they would add an element(banner/ad) to the page and don't care to wait on it?

Reply:
For now, I think Promise makes more sense. We could always change the type later if needed?

Reply:
Sure. Do we want to keep it that way when adding the interface to the API?

Reply:
That sounds good to me 👍

Reply:
Done. Copied the return type of `AudienceActionFlow.start`.